kyo-stm/shared/src/main/scala/kyo/TRefLog.scala

```diff
@@ -14,35 +17,26 @@ opaque type TRefLog = Map[TRef[Any], TRefLog.Entry[Any]]
 
 private[kyo] object TRefLog:
 
+    given tag: Tag[Var[TRefLog]] = Tag[Var[Map[TRef[Any], TRefLog.Entry[Any]]]]
+
     val empty: TRefLog = Map.empty
 
     extension (self: TRefLog)
 
         def put[A](ref: TRef[A], entry: Entry[A]): TRefLog =
-            self.updated(ref.asInstanceOf[TRef[Any]], entry.asInstanceOf[Entry[Any]])
+            val refAny   = ref.asInstanceOf[TRef[Any]]
+            val entryAny = entry.asInstanceOf[TRefLog.Entry[Any]]
+            self.updated(refAny, entryAny)
+        end put
 
         def get[A](ref: TRef[A]): Maybe[Entry[A]] =
             val refAny = ref.asInstanceOf[TRef[Any]]
             Maybe.when(self.contains(refAny))(self(refAny).asInstanceOf[Entry[A]])
 
-        def toSeq: Seq[(TRef[Any], Entry[Any])] =
-            self.toSeq
+        def toMap: Map[TRef[Any], TRefLog.Entry[Any]] = self
     end extension
 
-    def use[A, S](f: TRefLog => A < S)(using Frame): A < (S & Var[TRefLog]) =
```

Comment:
These methods were introducing a few allocations and triggering unnecessary boxing so I've changed `STM` to use `Var` directly and expose a `Tag[Var[TRefLog]]` (line 20)